src/backend/aspen/workflows/nextstrain_run/builder_base.py

```diff
@@ -29,28 +29,31 @@ def load_template(self):
 
     def update_build(self, config):
         build = config["builds"]["aspen"]
-        # TODO - We'll add region/country info to groups soon, but it's not there yet.
-        # region = self.group.region
-        # country = self.group.country
-        # build["region"] = region
-        # build["title"] = build["title"].format(division=division, location=location, country=country, region=region)
-        country = "USA"
-        division = self.group.division
-        location = self.group.location
-        build["country"] = country
-        build["division"] = division
-        build["location"] = location
+
+        location = self.group.default_tree_location
+        # Fill out region/country/division/location fields if the group has them,
+        # or remove those fields if they don't.
+        location_fields = ["region", "country", "division", "location"]
+        location_values = []
+        for field in location_fields:
+            value = getattr(location, field)
+            if value:
+                build[field] = value
+                location_values.append(value)
+            else:
+                del build[field]
 
         # NOTE: <BuilderClass>.subsampling_scheme is used in 3 places:
         #   - Its lowercase'd name is used to find a markdown file with an "about this tree" description
         #   - It refers to a subsampling_scheme key in the mega nextstrain template
         #   - It's title-case'd and included in the tree title as human-readable text
         build["subsampling_scheme"] = self.subsampling_scheme
-        build["title"] = build["title"].format(
+
+        # Update the tree's title with build type & location.
+        title_template = "{tree_type} tree for samples collected in {location}"
+        build["title"] = title_template.format(
             tree_type=self.subsampling_scheme.title(),
-            division=division,
-            location=location,
-            country=country,
+            location=", ".join(location_values),
```

Comment:
I think I understand this now. here we're specifically replacing the `{location}` in the `build["title"]` section, before writing out the `builds.yaml` for `ncov`

Reply:
yeah, but I changed it so now YAML builder is completely overwriting the title, and we don't have the {location} template in mega_template.yaml at all.